packages/medusa/src/services/payment-collection.ts

```diff
@@ -52,24 +80,89 @@ export default class PaymentCollectionService extends TransactionBaseService {
     )
 
     const query = buildQuery({ id: paymentCollectionId }, config)
-    const paymentCollection = await paymentCollectionRepository.findOne(query)
 
-    if (!paymentCollection) {
+    const paymentCollection = await paymentCollectionRepository.find(query)
+
+    if (!paymentCollection.length) {
       throw new MedusaError(
         MedusaError.Types.NOT_FOUND,
         `Payment collection with id ${paymentCollectionId} was not found`
       )
     }
 
-    return paymentCollection
+    return paymentCollection[0]
   }
 
-  async create(data: CreatePaymentCollectionInput): Promise<PaymentCollection> {
-    return await this.atomicPhase_(async (transactionManager) => {
-      const paymentCollectionRepository =
-        transactionManager.getCustomRepository(
-          this.paymentCollectionRepository_
+  async getPaymentCollectionIdBySessionId(
+    sessionId: string,
+    config: FindConfig<PaymentCollection> = {}
+  ): Promise<PaymentCollection> {
+    const manager = this.transactionManager_ ?? this.manager_
+    const paymentCollectionRepository = manager.getCustomRepository(
+      this.paymentCollectionRepository_
+    )
+
+    const paymentCollection = await paymentCollectionRepository.find({
+      join: {
+        alias: "payment_col",
+        innerJoin: { payment_sessions: "payment_col.payment_sessions" },
+      },
+      where: (qb) => {
+        qb.where(
+          "payment_col_payment_sessions.payment_session_id = :sessionId",
+          { sessionId }
         )
+      },
+      relations: config.relations,
+      select: config.select,
+    })
```

Comment:
suggestion(non-blocking): Refactor to a method in `PaymentCollectionRepository` to keep the service clean from querybuilding